lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Sequential.hs

```diff
@@ -197,30 +242,161 @@ deriveAddressPrivateKey
         addrXPrv = -- lvl5 derivation; soft derivation of address index
             deriveXPrv DerivationScheme2 pwd changeXPrv addrIx
     in
-        Key addrXPrv
+        SeqKey addrXPrv
 
 -- | Derives address public key from the given account public key, using
 -- derivation scheme 2 (see <https://github.com/input-output-hk/cardano-crypto/ cardano-crypto>
 -- package for more details).
 --
 -- This is the preferred way of deriving new sequential address public keys.
 deriveAddressPublicKey
-    :: Key 'AccountK XPub
+    :: SeqKey 'AccountK XPub
     -> ChangeChain
     -> Index 'Soft 'AddressK
-    -> Key 'AddressK XPub
-deriveAddressPublicKey (Key accXPub) changeChain (Index addrIx) =
+    -> SeqKey 'AddressK XPub
+deriveAddressPublicKey (SeqKey accXPub) changeChain (Index addrIx) =
     fromMaybe errWrongIndex $ do
         let changeCode = fromIntegral $ fromEnum changeChain
         changeXPub <- -- lvl4 derivation in bip44 is derivation of change chain
             deriveXPub DerivationScheme2 accXPub changeCode
         addrXPub <- -- lvl5 derivation in bip44 is derivation of address chain
             deriveXPub DerivationScheme2 changeXPub addrIx
-        return $ Key addrXPub
+        return $ SeqKey addrXPub
   where
     errWrongIndex = error $
         "Cardano.Wallet.Primitive.AddressDerivation.deriveAddressPublicKey \
         \failed: was given an hardened (or too big) index for soft path \
         \derivation ( " ++ show addrIx ++ "). This is either a programmer \
         \error, or, we may have reached the maximum number of addresses for \
         \a given wallet."
+
+
+{-------------------------------------------------------------------------------
+                                   Passphrase
+-------------------------------------------------------------------------------}
+
+-- | Re-encrypt a private key using a different passphrase.
+--
+-- **Important**:
+-- This function doesn't check that the old passphrase is correct! Caller is
+-- expected to have already checked that. Using an incorrect passphrase here
+-- will lead to very bad thing.
+changePassphraseSeq
+    :: Passphrase "encryption-old"
+    -> Passphrase "encryption-new"
+    -> SeqKey purpose XPrv
+    -> SeqKey purpose XPrv
+changePassphraseSeq (Passphrase oldPwd) (Passphrase newPwd) (SeqKey prv) =
+    SeqKey $ xPrvChangePass oldPwd newPwd prv
+
+{-------------------------------------------------------------------------------
+
+-------------------------------------------------------------------------------}
+
+-- | Extract the public key part of a private key.
+publicKeySeq
```

Comment:
I would reuse 
```
publicKey
    :: Key level XPrv
    -> Key level XPub
```
here

Reply:
ok, I have only now detected that `publicKey` does not take form like this anymore